lib/matplotlib/tests/test_dates.py

```diff
@@ -506,6 +506,57 @@ def _create_auto_date_locator(date1, date2):
         assert list(map(str, mdates.num2date(locator()))) == expected
 
 
+def test_concise_formatter():
+    def _create_auto_date_locator(date1, date2):
+        fig, ax = plt.subplots()
+
+        locator = mdates.AutoDateLocator(interval_multiples=True)
+        formatter = mdates.ConciseDateFormatter(locator)
+        ax.yaxis.set_major_locator(locator)
+        ax.yaxis.set_major_formatter(formatter)
+        ax.set_ylim(date1, date2)
+        fig.canvas.draw()
+        sts = []
+        for st in ax.get_yticklabels():
+            sts += [st.get_text()]
+        return sts
+
+    d1 = datetime.datetime(1997, 1, 1)
+    results = ([datetime.timedelta(weeks=52 * 200),
```

Comment:
Using test parametrization seems the better option here?

Reply:
Maybe, but all the other similar tests are written this way, and I', not going to change all of them, particularly given my lack of comfort with parameterized tests...

Reply:
It's a fairly straightforward rearrangement of the `for`-loop, essentially:
```python
@pytest.mark.parametrize(
    't_delta expected',
    # Here put the list/tuple of things that was in `results`.
)
def test_concise_formatter(t_delta, expected):
     d1 = datetime.datetime(1997, 1, 1)
     d2 = d1 + t_delta
    # And then put the rest of `_create_auto_date_locator`, but without the function.
```
You could do more fancy things with `ids` or `pytest.param`, but it's not really required here.

Reply:
That doesn't seem any more readable to me.   I *guess* there is some advantage to parameterizing tests from the point of view of running them all rather than having them fail sequentially, but....